ietf-layer0-types-ext.yang

```diff
@@ -51,4 +51,494 @@ module ietf-layer0-types-ext {
+  }
+  identity golay {
+    base FEC;
+      description "Golay error correction";
+  } 
+
+  identity line-coding {
+    description
+      "base line-coding class";
+    reference 
+      "ITU-T G.698.2-201811 section 7"   
+  }  
+
+  identity line-coding-NRZ-2p5G {
+    base line-coding;
+    description 
+      "ITU-T G.698.2-201811 section 7 table 8-1 "   
+  }
+
+  identity line-coding-NRZ-OTU1 {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-2 "   
+
+  }
+
+  identity line-coding-NRZ-10G {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-3/8-5 "   
+
+  }
+
+  identity line-coding-NRZ-OTU2 {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-4/8-6 "   
+
+  }
+
+// typedefs                
+
+  typedef operational-mode {
+    type string;
+    description
+      "Vendor-specific mode that guarantees 
+      interoperability.";
+    
+  }
+
+// temporary defined here for disalignment with 
+//ietf-layer0-types module
+  typedef standard-mode {
+    type string;
+    description
+      "ITU-T G.698.2 standard mode that guarantees 
+       interoperability.
+       It must be an string with the following format:
+       B-DScW-ytz(v) where all these attributes 
+       are conformant
+       to the ITU-T recomendation";
+    reference "ITU-T G.698.2 (11/2018)";
+  }
+
+  typedef frequency-thz {
+      type decimal64 {
+        fraction-digits 6;
+      }
+      units THz;
+      description
+        "The DWDM frequency in THz, e.g., 193.112500";
+      reference
+        "RFC6205: Generalized Labels for
+        Lambda-Switch-Capable (LSC) Label Switching Routers";
+    }
+
+// GROUPINGS  
+
+/* This grouping represent the list of capabilities attributes not under operational-mode */
+
+
+  grouping common-tranceiver-capabilities {
+    description "Capability of an optical tranceiver";
+
+ 
+/* transmitter tuning range (f_tx-min, f_tx-max) */
+
+    leaf min-central-frequency {
+      type layer0-types:frequency-thz;
+      config false;
+      description
+        "his parameter indicates the minimum frequency for
+                   this template.";
+    }
+    leaf max-central-frequency {
+      type layer0-types:frequency-thz;
+      config false;
+      description "This parameter indicates the minimum frequency
+                   for this template";
+    }
+
+    transmitter-tunability-grid {
+      type identityref {
+        base l0-types:flexi-ch-spc-type
+      }
+    }
+
+/* supported transmitter power range [p_tx-min, p_tx_max] */
+
+    leaf min-channel-output-power {
+      type dbm-t;
+      config false;
+      description "The minimum output power of this interface";
+    }
+    leaf max-channel-output-power {
+      type dbm-t;
+      config false;
+      description "The maximum output power of this interface";
+    }
+
+ /* supported receiver power range [p_rx-min, p_rx_max] */   

```

Comment:
G.698.2 application codes also defines the minimum and maximum mean input power to be supported (see clause 7.4.1 of G.698.2)

Reply:
**2020-08-13 CCAMP call**

@sergiobelotti : same resolution as in https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext/pull/4/files/840635ac33a7a32cddb5adb57b9862f0d2e3209e#r469993534